Error trying to login to root with zsh when user chronos runs zsh as shell

## Description

Due to this issue #6971 #6972 and this pull #6974

When I try to log in as root with zsh as the default shell via the chronos user running zsh, I get an error similar to that problem

In the chronos user running zsh I run:
```shell
sudo su
```
Or this:
```shell
sudo zsh
```

I got this error:
```
$ sudo su
su: error while loading shared libraries: libtinfo.so.6: cannot open shared object file: No such file or directory
```
and this:
```
$ sudo zsh
zsh: error while loading shared libraries: libgdbm.so.6: cannot open shared object file: No such file or directory
```

## Reproducing

1. Press ```Ctrl + Alt + T```, and type ```shell```
2. Get #6974 pull request's changes locally with ```CREW_TESTING_REPO=https://github.com/satmandu/chromebrew.git CREW_TESTING_BRANCH=zsh CREW_TESTING=1 crew update```
3. Install zsh with ```crew install zsh```
4. Change the root shell to zsh with ```sudo chsh -s $(which zsh)```
5. Run ```zsh```
6. Run ```sudo su``` or ```sudo zsh```


## Versions and system information
***
- Architecture: `x86_64` (`x86_64`)
- Kernel version: `5.10.106-brunch-sebanc`
- Chromebrew version: `1.23.11`
- Chromebrew prefix: `/usr/local`
- Chromebrew libdir: `/usr/local/lib64`
- Last update in local repository: `734ed5c: zsh -> 5.8.1`
- OS variant: `Chrome OS`
- OS version: `rammus-release/R100-14526.57.0`
- OS channel: `stable-channel`

CHROMEOS_ARC_ANDROID_SDK_VERSION=28
CHROMEOS_ARC_VERSION=8334839
CHROMEOS_AUSERVER=https://tools.google.com/service/update2
CHROMEOS_BOARD_APPID={625849FA-56A0-4E67-9163-B89BE0C2A6AE}
CHROMEOS_CANARY_APPID={90F229CE-83E2-4FAF-8479-E368A34938B1}
CHROMEOS_DEVSERVER=
CHROMEOS_RELEASE_APPID={625849FA-56A0-4E67-9163-B89BE0C2A6AE}
CHROMEOS_RELEASE_BOARD=rammus-signed-mp-v2keys
CHROMEOS_RELEASE_BRANCH_NUMBER=57
CHROMEOS_RELEASE_BUILDER_PATH=rammus-release/R100-14526.57.0
CHROMEOS_RELEASE_BUILD_NUMBER=14526
CHROMEOS_RELEASE_BUILD_TYPE=Official Build
CHROMEOS_RELEASE_CHROME_MILESTONE=100
CHROMEOS_RELEASE_DESCRIPTION=14526.57.0 (Official Build) stable-channel rammus 
CHROMEOS_RELEASE_KEYSET=mp-v2
CHROMEOS_RELEASE_NAME=Chrome OS
CHROMEOS_RELEASE_PATCH_NUMBER=0
CHROMEOS_RELEASE_TRACK=stable-channel
CHROMEOS_RELEASE_UNIBUILD=1
CHROMEOS_RELEASE_VERSION=14526.57.0
DEVICETYPE=CHROMEBOOK
GOOGLE_RELEASE=14526.57.0
***

When I try to recompile util_linux, I get an error:
```/tmp/ccj7w6Tb.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'crypt'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:7734: sulogin] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/usr/local/tmp/crew/util_linux.20220416015608.dir/util-linux-2.36.2'
make[1]: *** [Makefile:13798: all-recursive] Error 1
make[1]: Leaving directory '/usr/local/tmp/crew/util_linux.20220416015608.dir/util-linux-2.36.2'
make: *** [Makefile:5898: all] Error 2
/usr/local/lib/crew/lib/package.rb:258:in `system': Command failed with exit 2: bash
util_linux failed to install: `CFLAGS="-O2 -pipe -flto -ffat-lto-objects -fPIC -fuse-ld=mold " CXXFLAGS="-O2 -pipe -flto -ffat-lto-objects -fPIC -fuse-ld=mold " FCFLAGS="-O2 -pipe -flto -ffat-lto-objects -fPIC -fuse-ld=mold " FFLAGS="-O2 -pipe -flto -ffat-lto-objects -fPIC -fuse-ld=mold " LDFLAGS="-flto " bash -c make -j2` exited with 2
```

--- a/bin/crew
+++ b/bin/crew
@@ -1051,10 +1051,7 @@ def prepare_package(destdir)
     strip_dir destdir
 
     # Use patchelf to set need paths for all binaries.
-    # Currently broken with glibc on armv7l, but musl is fine.
-    unless %w[aarch64 armv7l].include?(ARCH) and !@pkg.is_musl?
-      patchelf_set_need_paths destdir
-    end
+    patchelf_set_need_paths destdir
 
     # use upx on executables
     shrink_dir destdir
@@ -1083,6 +1080,10 @@ def patchelf_set_need_paths(dir)
 
       @neededlibs.each_line(chomp: true) do |neededlibspatch|
         next if neededlibspatch.include?(@patchelf_lib_prefix.to_s)
+        # Avoid segfaults from not using system libdl.so
+        next if !@pkg.is_musl? && neededlibspatch.include?('libdl.so')
+        # Avoid breakage on armv7l
+        next if !@pkg.is_musl? && neededlibspatch.include?('ld-linux-armhf.so.3')
 
         @neededlib_basename = File.basename(neededlibspatch)
         @neededlibspatchednamepath = "#{@patchelf_lib_prefix}/" + @neededlib_basename

--- a/lib/const.rb
+++ b/lib/const.rb
@@ -1,6 +1,6 @@
 # Defines common constants used in different parts of crew
 
-CREW_VERSION = '1.23.11'
+CREW_VERSION = '1.23.12'
 
 ARCH_ACTUAL = `uname -m`.chomp
 # This helps with virtualized builds on aarch64 machines

--- a/packages/zsh.rb
+++ b/packages/zsh.rb
@@ -3,29 +3,30 @@ require 'package'
 class Zsh < Package
   description 'Zsh is a shell designed for interactive use, although it is also a powerful scripting language.'
   homepage 'http://zsh.sourceforge.net/'
-  version '5.8.1'
+  version '5.8.1-1'
   license 'ZSH and GPL-2'
   compatibility 'all'
   source_url 'https://downloads.sourceforge.net/project/zsh/zsh/5.8.1/zsh-5.8.1.tar.xz'
   source_sha256 'b6973520bace600b4779200269b1e5d79e5f505ac4952058c11ad5bbf0dd9919'
 
   binary_url({
-    aarch64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/zsh/5.8.1_armv7l/zsh-5.8.1-chromeos-armv7l.tar.zst',
-     armv7l: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/zsh/5.8.1_armv7l/zsh-5.8.1-chromeos-armv7l.tar.zst',
-       i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/zsh/5.8.1_i686/zsh-5.8.1-chromeos-i686.tar.zst',
-     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/zsh/5.8.1_x86_64/zsh-5.8.1-chromeos-x86_64.tar.zst'
+    aarch64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/zsh/5.8.1-1_armv7l/zsh-5.8.1-1-chromeos-armv7l.tar.zst',
+     armv7l: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/zsh/5.8.1-1_armv7l/zsh-5.8.1-1-chromeos-armv7l.tar.zst',
+       i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/zsh/5.8.1-1_i686/zsh-5.8.1-1-chromeos-i686.tar.zst',
+     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/zsh/5.8.1-1_x86_64/zsh-5.8.1-1-chromeos-x86_64.tar.zst'
   })
   binary_sha256({
-    aarch64: '66be508e6f844d36f2d173965b253231204f206d95785fe74a1c8e4280217126',
-     armv7l: '66be508e6f844d36f2d173965b253231204f206d95785fe74a1c8e4280217126',
-       i686: '579921c5ea4c7c66254f63b061463dcd2bcc32bc77a6ff5278c6a54f0d4e0777',
-     x86_64: 'c8c3e4cb8ec2a85684a134de076ed5dd8bf4dc36a067d03d68aba0073eacfe36'
+    aarch64: 'e2823b5b07a21ead7390305256fd04328f07d9f536bbf3bc0d77da54f9ace214',
+     armv7l: 'e2823b5b07a21ead7390305256fd04328f07d9f536bbf3bc0d77da54f9ace214',
+       i686: 'af70de10fa62dd86541934214a2addcbd20ae1a776b466bc90ad8f4e5b19cfbc',
+     x86_64: '1cf958108388929dc8743c85b095e17e4794407037ab058d579648593fdb425b'
   })
 
   depends_on 'gdbm' # R
   depends_on 'glibc' # R
   depends_on 'ncurses' # R
   depends_on 'gcc' # R
+  patchelf
 
   def self.build
     system "./configure #{CREW_OPTIONS}"

